### PR TITLE
[zh-cn] HTTP method updates

### PR DESCRIPTION
**Pull request by jasonren0403**
### Description

Review document under web/http/methods

### CODE REVIEW

File: files/zh-cn/web/http/methods/connect/index.md [hunk 1]
```diff
@@ -5,31 +5,55 @@ slug: Web/HTTP/Methods/CONNECT
 
 {{HTTPSidebar}}
 
-在 HTTP 协议中，**`CONNECT`** 方法可以开启一个客户端与所请求资源之间的双向沟通的通道。它可以用来创建隧道（tunnel）。
+**HTTP `CONNECT` 方法**可以开启与所请求资源之间的双向沟通的通道。它可以用来创建隧道（tunnel）。
 
-例如，**`CONNECT`** 可以用来访问采用了 {{Glossary("SSL")}} ({{Glossary("HTTPS")}}) 协议的站点。客户端要求代理服务器将 TCP 连接作为通往目的主机隧道。之后该服务器会代替客户端与目的主机建立连接。连接建立好之后，代理服务器会面向客户端发送或接收 TCP 消息流。
+例如，**`CONNECT`** 可以用来访问采用了 {{Glossary("SSL")}}（{{Glossary("HTTPS")}}）协议的站点。客户端要求 HTTP {{Glossary("Proxy
```
Comment (yin1999) [inline files/zh-cn/web/http/methods/connect/index.md:10]:
```suggestion
例如，**`CONNECT`** 可以用来访问采用了 {{Glossary("SSL")}}（{{Glossary("HTTPS")}}）协议的站点。客户端要求 HTTP {{Glossary("Proxy
```

File: files/zh-cn/web/http/methods/patch/index.md [hunk 2]
```diff
@@ -55,8 +86,8 @@ ETag: "e0023aa4f"
...
 {{Specifications}}
 
-## 相关内容
+## 参见
 
 - {{HTTPStatus("204")}}
-- {{HTTPHeader("Allow")}}, {{HTTPHeader("Access-Control-Allow-Methods")}}
-- {{HTTPHeader("Accept-Patch")}} – 用于明确服务器端可以接受的补丁文件的格式。
+- {{HTTPHeader("Allow")}}、{{HTTPHeader("Access-Control-Allow-Methods")}}
+- {{HTTPHeader("Accept-Patch")}}——指定了可以接受的补丁文件的格式。
```
Comment (yin1999) [inline files/zh-cn/web/http/methods/patch/index.md:93]:
```suggestion
- {{HTTPHeader("Accept-Patch")}}——指定了可以接受的补丁文档的格式。
```

File: files/zh-cn/web/http/methods/post/index.md [hunk 3]
```diff
@@ -5,56 +5,79 @@ slug: Web/HTTP/Methods/POST
 
 {{HTTPSidebar}}
 
-**HTTP `POST` 方法** 发送数据给服务器。请求主体的类型由 {{HTTPHeader("Content-Type")}} 首部指定。
+**HTTP `POST` 方法**发送数据给服务器。请求主体的类型由 {{HTTPHeader("Content-Type")}} 首部指定。
```
Comment (yin1999) [inline files/zh-cn/web/http/methods/post/index.md:8]:
```suggestion
**HTTP `POST` 方法**发送数据给服务器。请求主体的类型由 {{HTTPHeader("Content-Type")}} 标头指定。
```

File: files/zh-cn/web/http/methods/post/index.md [hunk 4]
```diff
@@ -5,56 +5,79 @@ slug: Web/HTTP/Methods/POST
...
+{{HTTPMethod("PUT")}} 和 `POST` 方法的区别是，`PUT` 方法是幂等的：调用一次与连续调用多次效果是相同的（即没有*副*作用），而连续调用多次相同的 `POST` 方法可能会有副作用，比如多次提交同一订单。
 
-一个 `POST` 请求通常是通过 [HTML 表单](/zh-CN/docs/Web/Guide/HTML/Forms)发送，并返回服务器的修改结果。在这种情况下，content type 是通过在 _{{HTMLElement("form")}} 元素中设置正确的 [`enctype`](/zh-CN/docs/Web/HTML/Element/form#enctype) 属性，或是在 {{HTMLElement("input") }} 和 {{HTMLElement("button")}} 元素中设置 [`formenctype`](/zh-CN/docs/Web/HTML/Element/input#formenctype) 属性来选择的_:
+一个 `POST` 请求通常是通过 [HTML 表单](/zh-CN/docs/Learn/Forms)发送的，并导致服务器的相应修改。在这种情况下，内容类型（content type）是通过在 {{HTMLElement("form")}} 元素中设置正确的 [`enctype`](/zh-CN/docs/Web/HTML/Element/form#enctype) 属性，或是在 {{HTMLElement("input") }} 和 {{HTMLElement("button")}} 元素中设置 [`formenctype`](/zh-CN/docs/Web/HTML/Element/input#formenctype) 属性来选择的：
 
-- `application/`_`x-www-form-urlencoded`: 数据被编码成以 `'&'` 分隔的键 - 值对，同时以 `'='` 分隔键和值。非字母或数字的字符会被 _{{glossary("percent-encoding")}}_: 这也就是为什么这种类型不支持二进制数据 (应使用 `multipart/form-data` 代替)._
-- _`multipart/form-data`_
-- _`text/plain`_
+- `application/x-www-form-urlencoded`：数据被编码成以 `'&'` 分隔的键——值对，同时以 `'='` 分隔键和值。键和值中非字母或数字的字符会被 [URL 编码](https://zh.wikipedia.org/wiki/百分号编码)，这也是这种类型不支持二进制数据的原因（应使用 `multipart/form-data` 代替）。
```
Comment (yin1999) [inline files/zh-cn/web/http/methods/post/index.md:14]:
```suggestion
- `application/x-www-form-urlencoded`：数据被编码成以 `'&'` 分隔的键值对，同时以 `'='` 分隔键和值。键和值中非字母或数字的字符会被 [URL 编码](https://zh.wikipedia.org/wiki/百分号编码)，这也是这种类型不支持二进制数据的原因（应使用 `multipart/form-data` 代替）。
```

File: files/zh-cn/web/http/methods/trace/index.md [hunk 5]
```diff
@@ -5,23 +5,44 @@ slug: Web/HTTP/Methods/TRACE
...
-| 有主体（body）的请求（request）          | 否  |
-| ---------------------------------------- | --- |
-| 包含主体（body）的成功的响应（response） | 否  |
-| {{Glossary("Safe/HTTP", "安全")}}             | 否  |
-| {{Glossary("Idempotent", "幂等")}}     | 是  |
-| {{Glossary("Cacheable", "可缓存")}}         | 否  |
-| 允许用于 HTML 表单（form）               | 否  |
+**HTTP `TRACE` 方法**沿着通往目标资源的路径进行信息回环测试，提供一个有用的调试机制。
+
+请求的最终接收者应将收到的信息（不包括下面描述的一些字段）作为 {{HTTPStatus("200")}}（`OK`）响应的信息主体反映给客户端，其 {{HTTPHeader("Content-Type")}} 为`message/http`。最终接收者是源服务器或第一个在请求中收到 {{HTTPHeader("Max-Forwards")}} 值为 0 的服务器。
```
Comment (yin1999) [inline files/zh-cn/web/http/methods/trace/index.md:10]:
```suggestion
请求的最终接收者应将收到的信息（不包括下面描述的一些字段）作为 {{HTTPStatus("200")}}（`OK`）响应的信息主体反映给客户端，其 {{HTTPHeader("Content-Type")}} 为 `message/http`。最终接收者是源服务器或第一个在请求中收到 {{HTTPHeader("Max-Forwards")}} 值为 0 的服务器。
```

File: files/zh-cn/web/http/methods/options/index.md [hunk 6]
```diff
@@ -26,60 +49,69 @@ OPTIONS * HTTP/1.1
...
+  - : {{HTTPMethod("POST")}}、{{HTTPMethod("GET")}} 和 `OPTIONS` 是该 URL 允许的方法（该标头类似于 {{HTTPHeader("Allow")}} 响应标头，但只用于 [CORS](/zh-CN/docs/Web/HTTP/CORS)）。
+- {{HTTPHeader("Access-Control-Allow-Headers")}}
+  - : `X-PINGOTHER` 和 `Content-Type` 是该 URL 允许的请求标头。
+- {{HTTPHeader("Access-Control-Max-Age")}}
+  - : 以上权限可以缓存 86400 秒（1 天）。
 
-```plain
-HTTP/1.1 200 OK
+```http
+HTTP/1.1 200 No Content
```
Comment (jasonren0403) [inline files/zh-cn/web/http/methods/options/index.md:99]:
```suggestion
HTTP/1.1 200 OK
```

File: files/zh-cn/web/http/methods/head/index.md [hunk 7]
```diff
@@ -5,23 +5,46 @@ slug: Web/HTTP/Methods/HEAD
...
-| 请求是否有正文                                         | 否  |
-| ------------------------------------------------------ | --- |
-| 成功的响应是否有正文                                   | 否  |
-| {{Glossary("Safe/HTTP", "安全")}}                           | 是  |
-| {{Glossary("Idempotent", "幂等")}}                           | 是  |
-| {{Glossary("Cacheable", "可缓存")}}                       | 是  |
-| [HTML 表单](/zh-CN/docs/Web/Guide/HTML/Forms) 是否支持 | 否  |
+**HTTP `HEAD` 方法**请求资源的[标头](/zh-CN/docs/Web/HTTP/Headers)信息，并且这些标头与 HTTP {{HTTPMethod("GET")}} 方法请求时返回的一致。该请求方法的一个使用场景是在下载一个大文件前先通过 `HEAD` 请求读取其 {{HTTPHeader("Content-Length")}} 标头的值获取文件的大小，而无需实际下载文件，以此可以节约带宽资源。
+
+> **警告：** `HEAD` 方法的响应*不应*包含响应体。即使包含了响应体，也*必须*忽略掉。响应体中的任何{{glossary("Representation header", "表示标头")}}若使用相似的 `GET` 请求也本应接收到。
```
[the remark this answers is not in the document]
Comment (jasonren0403) [reply files/zh-cn/web/http/methods/head/index.md:10]: 原句：If it has one anyway, that body must be ignored: any [representation headers](https://developer.mozilla.org/en-US/docs/Glossary/Representation_header) that might describe the erroneous body are instead assumed to describe the response which a similar GET request would have received.
你有什么建议吗？

**Review by yin1999 — APPROVED** [review]
Thanks

**Pull request merged**